src/coreclr/vm/jithelpers.cpp

```diff
@@ -634,72 +687,59 @@ HCIMPL1_V(UINT64, JIT_Dbl2ULngOvf, double val)
 }
 HCIMPLEND
 
-
-#if !defined(TARGET_X86) || defined(TARGET_UNIX)
-
-HCIMPL1_V(INT64, JIT_Dbl2Lng, double val)
+HCIMPL1_V(UINT32, JIT_Dbl2UInt, double val)
 {
     FCALL_CONTRACT;
+#if defined(TARGET_X86) || defined(TARGET_AMD64)
+    const double uint_max = 4294967295.0;
+    return (val > 0) ? ((val >= uint_max) ? UINT32_MAX : (UINT32)val) : 0;
```

Comment:
For consistency, could you please add "// Note that this expression also works properly for val = NaN case" comment to these conditions?

Reply:
*Added